lib/clearance/configuration.rb

```diff
@@ -112,7 +112,7 @@ def initialize
     end
 
     def user_model
-      @user_model ||= ::User
+      (@user_model || "User").to_s.constantize
```

Comment:
I'm not sure of the performance impact of calling `constantize` repeatedly (because `Clearance` references `user_model` a lot).

Perhaps something a little less-concise would be better:

```diff
diff --git a/lib/clearance/configuration.rb b/lib/clearance/configuration.rb
index d7c6863..ee0d54e 100644
--- a/lib/clearance/configuration.rb
+++ b/lib/clearance/configuration.rb
@@ -112,7 +112,11 @@ module Clearance
     end
 
     def user_model
-      (@user_model || "User").to_s.constantize
+      if @user_model.is_a?(String)
+        @user_model = (@user_model || "User").to_s.constantize
+      end
+
+      @user_model
     end
```

But in this version it feels awkward to be checking types.

Reply:
clearance is already calling `constantize` on every request from `reload_user_model` in this same file. Since this is called from a `to_prepare` block it will be invoked every time in dev mode but only once in production, which is fine. 

I think `user_model` could be left as-is. The custom user class could be given as a string already, and `reload_user_model` will convert it into a class before each request.

The only necessary thing is make sure clearance doesn't use the `User` class during initialization.

Reply:
That's an interesting suggestion - I tried it out quickly and it broke 48 / 198 examples - so I think the testing setup becomes a bit more involved (we'd have to call reload before each spec perhaps).

I do like the suggestion overall, but I think it's simpler to update `def user_model` because it's all self-contained - you can read it and understand what's going on, without having to know about the `reload_user_model` mechanism.